### ISSUE TITLE
Learning rate shows an abnormal changing trend and AP =0 while training queryinst model with my own dataset 

### ISSUE THREAD

**Issue by zxw0919** [hustvl/QueryInst#21]
Hi,

Thanks for your work!

Recently, I met two issues while training 'queryinst_r50_fpn_1x_coco'model with my own dataset.(samples_per_gpu=2,workers_per_gpu=2,
optimizer = dict(type='AdamW', lr=2.5e-05, weight_decay=0.0001) just as default settings)

I noticed that learning rate represents an abnormal changing trend: 
1. Starting with **INCREASINGING** trend in the first epoch ( all values are greater than the default lr setting value which is 2.5e-5)
2. Keeping decreasing to and then stay with the default lr setting value (2.5e-5) for next few epochs.
3. Decreasing again and stick to a value, for example, 2.5e-07.

This trending way is abnormal compared with some conventional lr trending patterns in which lr usually stays still or keeps deceasing during the training process. 

The second issue I met is all the AP and AR values equal to zero all the time. I attached the training log here for review.
[20210819_092034.log](https://github.com/hustvl/QueryInst/files/7011379/20210819_092034.log)

Could you help me with this? Thanks a lot!

**Comment by vealocia** [comment]
The first issue is normal for QueryInst. QueryInst is trained with two learning rate adjustment strategies: ```warmup``` and ```step down```. The ```warmup``` strategy provides an additive learning rate for the first few iterations (~1K iters) and is set up [here](https://github.com/hustvl/QueryInst/blob/1cca85154539c1c665cf34db61e60a77b27763c6/configs/queryinst/queryinst_r50_fpn_1x_coco.py#L124). The ```step down``` strategy decreases learning at certain epochs and it was set up [here](https://github.com/hustvl/QueryInst/blob/1cca85154539c1c665cf34db61e60a77b27763c6/configs/queryinst/queryinst_r50_fpn_100_proposals_mstrain_480-800_3x_coco.py#L22). So in the total training process, learning rate will firstly increase to the default setting within ~1k iterations and then decrease by 10 at 27th and 33rd epochs.
For the second issue, the training log shows that there seems no problem during your training process. So I suggest you check out your evaluation and inference code to find the way out.

**Comment by zxw0919** [comment]
@vealocia  Thanks for your comment :) !  I do have a deeper understanding for the learning rate adjustment strategies of QueryInst. That helps a lot.

However, for the second issue, I still can't figure it out. My evaluation and inference code remain as the original version of GitHub repository, I didn't change anywhere of them after I downloaded the code from Github. Based on a lot of experiments,  I noticed that : Bbox AP and Bbox AR  usually perform well with high MAP values, but Segms AP and AR still keep as zeros all the time. 

For example: 2021-08-25 12:30:34,516 - mmdet - INFO - Epoch(val) [30][385]	bbox_mAP: 0.8950, bbox_mAP_50: 1.0000, bbox_mAP_75: 0.9650, bbox_mAP_s: 0.7940, bbox_mAP_m: 0.9260, bbox_mAP_l: 0.9100, bbox_mAP_copypaste: 0.895 1.000 0.965 0.794 0.926 0.910, segm_mAP: 0.0000, segm_mAP_50: 0.0000, segm_mAP_75: 0.0000, segm_mAP_s: 0.0000, segm_mAP_m: 0.0000, segm_mAP_l: 0.0000, segm_mAP_copypaste: 0.000 0.000 0.000 0.000 0.000 0.000


I attached the whole log below just in case. Thanks!
[20210825_090242.log](https://github.com/hustvl/QueryInst/files/7044052/20210825_090242.log)
